engine/apps/user_management/models/organization.py

```diff
@@ -50,22 +53,37 @@ def create(self, **kwargs):
         return instance
 
     def delete(self):
-        org_id = self.public_primary_key
-        super().delete(self)
-        if settings.FEATURE_MULTIREGION_ENABLED:
-            delete_oncall_connector_async.apply_async(
-                (org_id),
-            )
+        self.update(deleted_at=timezone.now())
+
+    def hard_delete(self):
```

Comment:
why is the `hard_delete` method needed on both the `OrganizationQuerySet` and `Organization` classes?

Reply:
because it can be called both from queryset and object